arangod/Pregel/Actor/include/Actor/ActorList.h

```diff
@@ -0,0 +1,133 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2022 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Markus Pfeiffer
+/// @author Julia Volmer
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "Actor/ActorBase.h"
+#include "Actor/ActorPID.h"
+#include "Basics/Guarded.h"
+
+namespace arangodb::pregel::actor {
+
+struct ActorList {
+  struct ActorMap : std::unordered_map<ActorID, std::shared_ptr<ActorBase>> {};
+
+  ActorList(ActorMap map) : actors{std::move(map)} {}
+  ActorList() = default;
+
+  auto find(ActorID id) -> std::optional<std::shared_ptr<ActorBase>> {
+    return actors.doUnderLock(
+        [id](ActorMap& map)
+            -> std::optional<
+                std::reference_wrapper<std::shared_ptr<ActorBase>>> {
+          auto a = map.find(id);
+          if (a == std::end(map)) {
+            return std::nullopt;
+          }
+          auto& [_, actor] = *a;
+          return actor;
+        });
+  }
+
+  auto add(ActorID id, std::shared_ptr<ActorBase> actor) -> void {
+    actors.doUnderLock([id, actor = std::move(actor)](ActorMap& map) {
+      map.emplace(id, std::move(actor));
+    });
+  }
+
+  auto remove(ActorID id) -> void {
+    auto actor = find(id);
+    if (actor.has_value()) {
+      actors.doUnderLock([id](ActorMap& map) { map.erase(id); });
+    }
+  }
+
+  auto removeIf(
+      std::function<bool(std::shared_ptr<ActorBase> const&)> deletable)
+      -> void {
+    actors.doUnderLock([deletable = std::move(deletable)](ActorMap& map) {
+      std::erase_if(map, [deletable = std::move(deletable)](const auto& item) {
+        auto& [_, actor] = item;
+        return deletable(actor);
+      });
+    });
```

Comment:
I think this could be simplified a bit:
```suggestion
    actors.doUnderLock([&deletable](ActorMap& map) {
      std::erase_if(map, [&deletable](const auto& item) {
        return deletable(actor->second);
      });
    });
```
By capturing by reference we avoid move-construction and destruction of a few additional shared_ptr instances (micro-optimization).